src/api/democracy.ts

```diff
@@ -28,15 +34,28 @@ export const useReferendumInfo = (referendumIndex: string) => {
   )
 }
 
-export const getReferendums = (api: ApiPromise) => async () => {
-  const res = await api.query.democracy.referendumInfoOf.entries()
-  const referendums = res.map(([key, codec]) => ({
-    id: key.args[0].toString(),
-    referendum: codec.unwrap(),
-  }))
+export const getReferendums =
+  (api: ApiPromise, accountId?: string) => async () => {
+    const [referendumRaw, votesRaw] = await Promise.all([
+      api.query.democracy.referendumInfoOf.entries(),
+      accountId ? api.query.democracy.votingOf(accountId) : undefined,
+    ])
 
-  return referendums
-}
+    const referendums = referendumRaw.map(([key, codec]) => {
+      const id = key.args[0].toString()
+      const vote = votesRaw?.asDirect.votes.find(
+        (vote) => vote[0].toString() === id,
+      )
```

Comment:
[Array.some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) will be a better choice in this case.